File: src/test/java/com/ragin/bdd/cucumbertests/library/test/BodyManipulation.java
package com.ragin.bdd.cucumbertests.library.test;

import java.util.List;
import lombok.Getter;
import lombok.Setter;
import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.lang.NonNull;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class BodyManipulation {
    @PostMapping("/api/v1/body/manipulate")
    public ResponseEntity<String> stubBodyManipulation(
            final @NonNull @RequestBody StubBody body
    ) {
        return ResponseEntity.status(HttpStatus.CREATED).body(createManipulatedBodyResponse(body));
    }

    private String createManipulatedBodyResponse(@NonNull final StubBody body) {
        final JSONObject jsonObject = new JSONObject();
        try {
            final JSONArray newIdsArray = new JSONArray();
            body.ids.forEach(newIdsArray::put);

            jsonObject.put("newName", body.getName());
            jsonObject.put("newIds", newIdsArray);
        } catch (JSONException je) {
        }
        return jsonObject.toString();
    }

    @Getter
    @Setter
    public static class StubBody {
        private String name;
        private List<String> ids;
    }
}

File: src/test/kotlin/com/ragin/bdd/cucumbertests/library/test/BodyManipulation.kt
package com.ragin.bdd.cucumbertests.library.test

import com.ragin.bdd.cucumber.utils.JacksonUtils
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class BodyManipulation {
    @PostMapping("/api/v1/body/manipulate")
    fun stubBodyManipulation(
        @RequestBody body: StubBody
    ): ResponseEntity<String> {
        return ResponseEntity.status(HttpStatus.CREATED).body(createManipulatedBodyResponse(body))
    }

    private fun createManipulatedBodyResponse(body: StubBody): String {
        return JacksonUtils.mapper.writeValueAsString(
            mapOf(
                "newName" to body.name,
                "newIds" to body.ids
            )
        )
    }

    data class StubBody(
        val name: String? = null,
        val ids: List<String>
    )
}
